### ISSUE TITLE
Links to eBenefits should be made SSOe-aware

### ISSUE THREAD

**Issue by patrickvinograd** [department-of-veterans-affairs/va.gov-team#7972]
## User Story or Problem Statement
As a user, I need to be linked to eBenefits in a way that preserves my SSOe session so I can arrive at eBenefits in a logged-in state and have a seamless experience.

## Goal
* VA.gov is integrated sign-in with SSOe - when a user logs in they will have an SSOe session established.
* eBenefits is similarly further integrating with SSOe, and as of ~May 1 will accept the same three credential types that VA.gov does.
* But, in order for a VA.gov user to arrive at eBenefits in a way that will use their SSOe session, we need to update all the links to eBenefits to use the "standard junction" URL mechanism. Links that used to look like `www.benefits.va.gov/foo` will instead look something like `eauth.va.gov/ebenefits/foo`. This should be done for all "authenticated" links/CTAs - it probably does not need to occur for links that occur in general static content on the unauthenticated parts of VA.gov.
* eBenefits will only allow users in via SSO if they have a verified account  - ID.me identity proofed, or MHV premium or DSLogon Level 2, so if the authenticated links/CTAs aren't already gated by users having verified accounts, then we would additionally want to prompt them to verify before navigating them to eBenefits. 

## Objectives or Key Results this is meant to further
- _Include references to KPIs we intend to impact_
---
## Tasks
- [ ] Identify list of all authenticated-state links/CTAs to eBenefits. Provide to eBenefits team to determine mapping to SSOe-enabled URL structure.
- [ ] Implement code to conditionalize the links to eBenefits based on whether an SSOe session is present. See similar work in #7237 for MHV links - can likely build off that work other than where it uses the MHV base URL. 
- [ ] Update any content that refers to links to eBenefits only working with DSLogon, which will no longer be the case as of May 1 2020.
- [ ] Verify whether all authenticated-state links require LOA3 already; if not, implement a prompt to users to verify identity before linking them. (If this turns into implementation work, probably spawn it off as its own ticket.

## Acceptance Criteria
- [ ] Users who sign in to VA.gov and navigate to eBenefits will arrive already signed in, for all three credential options.
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by patrickvinograd** [comment]
This ticket needs assigning, working with @steve-gov to figure out who will work on this.

**Comment by steve-gov** [comment]
First pass at a list here: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/teams/ebenefits/research/ebenefits-sso-links.md

**Comment by steve-gov** [comment]
I think this list is good enough to get started. FWIW we could change the few links that are not in the `https://www.ebenefits.va.gov/ebenefits/about/feature?feature=XYZ` pattern, if having every link conform to that pattern provides a quicker or more robust solution.

**Comment by patrickvinograd** [comment]
Update  from eBenefits team - they are analyzing the above list to ensure that they have viable eauth-enabled versions. 

But, their May 1 SSOe launch is at risk, and they don't want us sending users over to the eauth-based URLs until they are ready to launch. So when implemented, this should be put behind a feature flag so we can turn it on in coordination with eBenefits (and also to enable testing it selectively in the run up to their launch).

**Comment by patrickvinograd** [comment]
@ericbuckley the pattern/mapping I provided was speculative...I think you can make headway on this but the exact target URLs are still pending confirmation from the eBenefits team.
